src/memray/_memray/sink.cpp

```diff
@@ -24,6 +24,19 @@ using namespace memray::exception;
 
 namespace {  // unnamed
 
+#ifdef __APPLE__
+static int
+posix_fallocate(int fd, off_t offset, off_t len)
+{
+    fstore_t store = {F_ALLOCATEALL, F_PEOFPOSMODE, 0, len, 0};
+    int res = fcntl(fd, F_PREALLOCATE, &store);
+    if (res != 0) {
+        return -1;
+    }
+    return ftruncate(fd, offset + len);
```

Comment:
The real `posix_fallocate` returns an error number, instead of setting errno, and our code that calls it checks that. I think this needs to be:
```cpp
    if (res != 0) {
        return errno;
    }
    res = ftruncate(fd, offset + len);
    if (res != 0) {
        return errno;
    }
    return 0;
```

But also, we probably need to retry on `EINTR` here. Our code that's calling `posix_fallocate` already loops if it gets `EINTR` returned to it, but that's not enough here - if `ftruncate` is repeatedly interrupted but `fcntl` succeeds on every iteration, we'll wind up preallocating far more space than we need.

Reply:
Done in a fixup